metabase/metabase
# feat(sdk): use public-facing question type in event handlers

Closes https://github.com/metabase/metabase/issues/50487, Closes https://github.com/metabase/metabase/issues/50505

### Description

Introduces the public-facing `MetabaseQuestion` type with transforms, and updates the existing event handlers to use the public metabase question type.

### How to verify

- Renders the Question and click on "Save"
- The `onSave` and `onBeforeSave` handlers should now give you the `MetabaseQuestion` type, instead of the internal type.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - Unit and E2E
